image/cli/mascli/functions/gitops_db2u_database

```diff
@@ -576,19 +592,43 @@ DB2_WORKLOAD: '${DB2_WORKLOAD}'"
     check_argo_app_synced mas-${MAS_INSTANCE_ID}-${MAS_APP_ID}-db2-database ${CLUSTER_ID}-watcher ${DB2_NAMESPACE}
     check_argo_app_healthy mas-${MAS_INSTANCE_ID}-${MAS_APP_ID}-db2-database ${CLUSTER_ID}-watcher
   fi
-  # Run specific setupcmds for Manage only if we have made changes.
-  if [[ "$DB2_INTERNAL" == "true" && "$FILES_ADDED" != "0" && "$MAS_APP_ID" == "manage" ]]; then
-    invoke_suite_db2_setup_for_manage 
-  fi
-  # Run DB2 apply script if we have made any changes. Needs to be run twice for DB2 Standalone as that is what ansible-devops role does
-  if [[ "$DB2_INTERNAL" == "true" && "$FILES_ADDED" != "0" ]]; then
+  echo "DB2CFG_SETTINGS_APPLIED= ${DB2CFG_SETTINGS_APPLIED} FILES_ADDED= ${FILES_ADDED} DB2_INTERNAL=${DB2_INTERNAL} MAS_APP_ID=${MAS_APP_ID}"
+
+  if [[ "$DB2_INTERNAL" == "true" && ("${DB2CFG_SETTINGS_APPLIED}" != "true" || "${FILES_ADDED}" != "0") ]]; then
+    
+    # Run specific setupcmds for Manage only if we have made changes.
+    if [[ "$MAS_APP_ID" == "manage" ]]; then
+      echo "invoke_suite_db2_setup_for_manage ..."
+      invoke_suite_db2_setup_for_manage 
+    fi
+    
+    # Run DB2 apply script if we have made any changes. Needs to be run twice for DB2 Standalone as that is what ansible-devops role does
+    echo "oc exec /db2u/scripts/apply-db2cfg-settings.sh ..."
     oc exec -n ${DB2_NAMESPACE} c-${DB2_INSTANCE_NAME}-db2u-0 -- su -lc '/db2u/scripts/apply-db2cfg-settings.sh | tee /tmp/apply-db2cfg-settings.log' db2inst1
     rc=$?
     [ $rc -ne 0 ] && exit $rc
     oc exec -n ${DB2_NAMESPACE} c-${DB2_INSTANCE_NAME}-db2u-0 -- su -lc '/db2u/scripts/apply-db2cfg-settings.sh | tee /tmp/apply-db2cfg-settings.log' db2inst1
     rc=$?
     [ $rc -ne 0 ] && exit $rc
+
+
+    if [[ "${DB2CFG_SETTINGS_APPLIED}" != "true" ]]; then
+      echo "sm_update_secret ...DB2CFG_SETTINGS_APPLIED with true"
+      export DB2CFG_SETTINGS_APPLIED="true"
+      sm_update_secret ${DB2CFG_SETTINGS_APPLIED_SECRET} "{ \"db2cfg_settings_applied\": \"${DB2CFG_SETTINGS_APPLIED}\" }"
+
+      echo
+      echo_reset_dim "- Re-generating DB2U Database file ${GITOPS_APPS_DIR}/${MAS_INSTANCE_ID}.${MAS_APP_ID}.ibm-db2u-database.yaml"
```

Comment:
As we are updating a secret in secrets manager and using that to determine whether the settings have been applied or not then I think we should remove the change to "Re-generating DB2U Database file" as we don't need it in two places. Either we put it in the secretsmanager or we put it in the helm values file but not both.